step/output.go

```diff
@@ -0,0 +1,80 @@
+package step
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/bitrise-io/go-steputils/output"
+	"github.com/bitrise-io/go-utils/v2/command"
+	"github.com/bitrise-io/go-utils/v2/env"
+)
+
+type OutputExporter interface {
+	ZipAndExportOutput(artifact, destinationZipPth, envKey string) error
+	CopyAndSaveTestData(artifact, targetAddonPath, testName string) error
+}
+
+type outputExporter struct {
+}
+
+func NewOutputExporter() OutputExporter {
+	return outputExporter{}
+}
+
+func (e outputExporter) ZipAndExportOutput(artifact, destinationZipPth, envKey string) error {
+	return output.ZipAndExportOutput([]string{artifact}, destinationZipPth, envKey)
+}
+
+func (e outputExporter) CopyAndSaveTestData(artifact, targetAddonPath, testName string) error {
+	testName = replaceUnsupportedFilenameCharacters(testName)
+	addonPerStepOutputDir := filepath.Join(targetAddonPath, testName)
+
+	if err := copyDirectory(artifact, addonPerStepOutputDir); err != nil {
+		return err
+	}
+	if err := saveBundleMetadata(addonPerStepOutputDir, testName); err != nil {
+		return err
+	}
+	return nil
+}
+
+// Replaces characters '/' and ':', which are unsupported in filnenames on macOS
+func replaceUnsupportedFilenameCharacters(s string) string {
+	s = strings.Replace(s, "/", "-", -1)
+	s = strings.Replace(s, ":", "-", -1)
+	return s
+}
```

Comment:
This feels like it would be a good candidate for some sort of library. Do we have a good place for something like this in one of our libraries?

Reply:
That seems like a good spot for it. If we add it to a library, we also need to ensure that this handles Linux invalid characters as well — assuming they are different than macOS ones. I'm not sure if they are.

Reply:
I would keep it here for now.
Looking at Pathmodifier again, maybe it is not the best place for this func.
All Pathmodifier funcs work with an external state, while ReplaceUnsupportedFilenameCharacters is a simple string modification, no need to break this dependency + I'm not sure if anyone else would need this logic.